client/tinychain/ml/linalg.py

```diff
@@ -269,7 +268,67 @@ def step(i: UInt):
     return After(Stream.range((0, n)).for_each(step), Tuple([cxt.sign_result, cxt.logdet_result]))
 
 
-def svd(matrix: Tensor) -> Tuple:
-    """Return the singular value decomposition of the given `matrix`."""
+@post_op
+def svd(cxt, A: Tensor, l=UInt(0), epsilon=F32(1e-5), max_iter=UInt(1000)) -> Tuple:
```

Comment:
Instead of `F32` this should just be `Float` (same below) because it might be called with a `Tensor` with dtype `F64` as well